include/aws/common/string.h

```diff
@@ -91,13 +135,29 @@ AWS_STATIC_IMPL bool aws_string_eq_byte_cursor(const struct aws_string *str, con
  */
 AWS_STATIC_IMPL
 bool aws_string_eq_byte_cursor_ignore_case(const struct aws_string *str, const struct aws_byte_cursor *cur) {
+    AWS_PRECONDITION(!str || aws_string_is_valid(str));
+    AWS_PRECONDITION(!cur || aws_byte_cursor_is_valid(cur));
+    if (str == NULL && cur == NULL) {
```

Comment:
This code is repeated by multiple functions.  I wonder if there is a macro that could be used here?

Reply:
I didn't find any macro available now to replace it, but we can try to come up with a new one in another PR.